vaadinfrontend/src/main/java/life/qbic/datamanager/views/navigation/ProjectSideNavigationComponent.java

```diff
@@ -0,0 +1,206 @@
+import life.qbic.datamanager.views.Context;
+import life.qbic.datamanager.views.projects.overview.ProjectOverviewPage;
+import life.qbic.datamanager.views.projects.project.ProjectMainLayout;
+import life.qbic.datamanager.views.projects.project.info.ProjectInformationMain;
+import life.qbic.projectmanagement.application.ExperimentInformationService;
+import life.qbic.projectmanagement.application.ProjectInformationService;
+import life.qbic.projectmanagement.application.ProjectPreview;
+import life.qbic.projectmanagement.application.SortOrder;
+import life.qbic.projectmanagement.domain.project.Project;
+import life.qbic.projectmanagement.domain.project.ProjectId;
+import life.qbic.projectmanagement.domain.project.experiment.Experiment;
+import life.qbic.projectmanagement.domain.project.experiment.ExperimentId;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Project Side Navigation Component
+ * <p>
+ * Allows the user to switch between the components shown in each {@link ProjectMainLayout} by
+ * clicking on the corresponding {@link SideNavItem} within Component which routes the user to the
+ * respective route defined in {@link life.qbic.datamanager.views.AppRoutes} for the component in
+ * question.
+ */
+
+@SpringComponent
+@UIScope
+public class ProjectSideNavigationComponent extends Div implements
+    BeforeEnterObserver {
+
+  private static final Logger log = getLogger(ProjectSideNavigationComponent.class);
+  private final Div projectSection = new Div();
+  private final SideNavItem projectSectionHeader = new SideNavItem("");
+  private final Div experimentSection = new Div();
+  private final SideNavItem experimentSectionHeader = new SideNavItem("");
+  private final transient ProjectInformationService projectInformationService;
+  private final transient ExperimentInformationService experimentInformationService;
+  public static final String PROJECT_ID_ROUTE_PARAMETER = "projectId";
+  public static final String EXPERIMENT_ID_ROUTE_PARAMETER = "experimentId";
+  private transient Context context = new Context();
+  private final MenuBar projectSelectMenu = new MenuBar();
+
+  public ProjectSideNavigationComponent(
+      @Autowired ProjectInformationService projectInformationService,
+      @Autowired ExperimentInformationService experimentInformationService) {
+    Objects.requireNonNull(projectInformationService);
+    Objects.requireNonNull(experimentInformationService);
+    this.addClassName("project-navigation-drawer");
+    this.projectInformationService = projectInformationService;
+    this.experimentInformationService = experimentInformationService;
+    projectSection.addClassName("project-section");
+    experimentSection.addClassName("experiment-section");
+    add(projectSection, experimentSection);
+    log.debug(
+        "New instance for ProjectSideNavigationComponent {} was created",
+        System.identityHashCode(this));
```

Comment:
Why is this necessary to log?

Reply:
We log all our component initialization currently. I think it makes sense for debugging purposes to see if the components could be created correctly 🤔